### PR TITLE
go-jet: init at 2.9.0

### PR DESCRIPTION
**Pull request by mrityunjaygr8**
###### Description of changes
[go-jet](https://github.com/go-jet/jet): Type safe SQL builder with code generation and automatic query result data mapping 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by JulienMalka — CHANGES_REQUESTED** [review]
Hello, thanks for your contribution. Are these 3 packages related to one another ? Otherwise could you split this into several PRs ?

**Comment by mrityunjaygr8** [comment]
My apologies for clubbing the packages in one branch. I needed them together for a project, and initially did not know how to use the nix files directly in the nix shell.

In the next couple of days i will split them into their own dedicated PR.

In the meantime,I will convert this PR into a draft.

**Review by JulienMalka — CHANGES_REQUESTED** [review]
Could you split this change into 2 commits : 
- ``maintainers: add mrityunjaygr8``
- ``go-jet: init at 2.9.0``
Thanks :)

File: pkgs/development/tools/go-jet/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    sha256 = "sha256-1kJaBLZIunexjxjOy55Nw0WMEhrSu+ptMbWVOJ1e5iA=";
+  };
+
+  vendorSha256 = "sha256-mhH9P3waINZhP+jNg3zKlssIL1ZO5xOBHp19Bzq/pSQ=";
+
+  subPackages = [ "cmd/jet" ];
+
+  tags = ["mysql" "golang" "postgres" "sql" "database" "code-generator" "sqlite" "postgresql" "mariadb" "sql-query" "codegenerator" "typesafe" "sql-builder" "datamapper" "code-completion" "sql-queries" "cockroachdb" "sql-query-builder" "sqlbuilder" "typesafety" ];
+
+  # doCheck = false;
```
Comment (JulienMalka) [inline pkgs/development/tools/go-jet/default.nix:20]:
```suggestion
```

File: pkgs/development/tools/go-jet/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
+{ lib, buildGoModule, fetchFromGitHub }:
+
+buildGoModule rec {
+  pname = "go-jet";
+  version = "2.9.0";
+
+  src = fetchFromGitHub {
+    owner = "go-jet";
```
Comment (JulienMalka) [inline pkgs/development/tools/go-jet/default.nix:8]:
```suggestion
    owner = pname;
```

**Review by JulienMalka — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/go-jet/default.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    repo = "jet";
+    rev = "v${version}";
+    sha256 = "sha256-1kJaBLZIunexjxjOy55Nw0WMEhrSu+ptMbWVOJ1e5iA=";
+  };
+
+  vendorSha256 = "sha256-mhH9P3waINZhP+jNg3zKlssIL1ZO5xOBHp19Bzq/pSQ=";
+
+  subPackages = [ "cmd/jet" ];
+
+  tags = ["mysql" "golang" "postgres" "sql" "database" "code-generator" "sqlite" "postgresql" "mariadb" "sql-query" "codegenerator" "typesafe" "sql-builder" "datamapper" "code-completion" "sql-queries" "cockroachdb" "sql-query-builder" "sqlbuilder" "typesafety" ];
```
Comment (SuperSandro2000) [inline pkgs/development/tools/go-jet/default.nix:18]:
```suggestion
  tags = [
    "mysql"
    "golang"
    "postgres"
    "sql"
    "database"
    "code-generator"
    "sqlite"
    "postgresql"
    "mariadb"
    "sql-query"
    "codegenerator"
    "typesafe"
    "sql-builder"
    "datamapper"
    "code-completion"
    "sql-queries"
    "cockroachdb"
    "sql-query-builder"
    "sqlbuilder"
    typesafety"
  ];
```

we don't really have a line length but thats a bit long

File: pkgs/development/tools/go-jet/default.nix [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    "datamapper"
+    "code-completion"
+    "sql-queries"
+    "cockroachdb"
+    "sql-query-builder"
+    "sqlbuilder"
+    "typesafety"
+  ];
+
+  preBuild = ''
```
Comment (NickCao) [inline pkgs/development/tools/go-jet/default.nix:41]: Do this in postPatch is more appropriate.

File: pkgs/development/tools/go-jet/default.nix [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    # removing the tests which depend on external data
+    rm -rf tests/sqlite
+    rm -rf tests/postgres
+    rm -rf tests/mysql
```
Comment (NickCao) [inline pkgs/development/tools/go-jet/default.nix:45]:
```suggestion
    rm -r tests/{sqlite,postgres,mysql}
```

**Review by NickCao — APPROVED** [review]

**Pull request merged**